### PR TITLE
[OSC-1136, OSC-1137] - add 'compare' and 'complete' commands

### PR DESCRIPTION
**Pull request by ChintanRaval**
Adds the below new commands:
- `compare`: Compares & persists SHA256-hashed checksums of the given models against those of the last successful execution. Returns space-separated string of changed model names. Parameters required:
  - `execution-id`: a GUID identifier of an existing data pipeline execution as returned by the `init` command.
  - `model-type`: type of models being processed e.g.: `load`, `transform`, etc. this `model-type` is used to group the model checksums by and used to find and compare older ones.
  - `base-path`: absolute or relative path to the models e.g.: `./load`, `/home/local/load`, `C:/path/to/load`
  - `model-patterns`: path-based patterns _(relative to `base-path`)_ to different models with extensions. models within a model-type must be named uniquely regardless of their file extension. e.g.: `*.txt`, `**/*.txt`, `./relative/path/to/some_models/**/*.csv`, `relative/path/to/some/more/related/models/**/*.sql`
- `complete`: Marks the completion of an existing execution by updating a record for the same in the given database. Returns nothing unless there's an error. Parameter required:
  - `execution-id`: a GUID identifier of an existing data pipeline execution as returned by the `init` command.

### CODE REVIEW

**Comment by ChintanRaval** [comment]
I have tried a few approached before ending up with the one in this PR but I am not content with what I've done here... so I'll be doing more reading over the best practices before deciding on whether or not to go further with the current approach. 🤔

**Comment by elexisvenator** [comment]
> _No description provided._

???

File: README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ py mcd.py [options] <command> [command-parameters]
   - `finish`: Marks the completion of an existing execution by updating a record for the same in the given database. Returns nothing unless there's an error.
     - `db-connection-string`: a [PostgreSQL Db Connection String](http://docs.sqlalchemy.org/en/latest/dialects/postgresql.html#module-sqlalchemy.dialects.postgresql.psycopg2) of the format `postgresql+psycopg2://user:password@host:port/dbname`
     - `execution-id`: a GUID identifier of an existing data pipeline execution.
+    - `models_folder_path`: absolute or relative path to models e.g.: `./models`
```
Comment (waltaro) [inline README.md]:
```suggestion
    - `models-folder-path`: absolute or relative path to models e.g.: `./models`
```

File: modules/commands/FinishCommand.py [hunk 2]
```diff
@@ -1,11 +1,40 @@
...
+        self._models_folder_path = models_folder_path
+        self._models_file_pattern = '*.sql'
 
     def execute(self):
-        data_pipeline_execution = self.repository.finish_existing(self._executionId)
-        self.logger.debug('Finised data_pipeline_execution = ' + str(data_pipeline_execution))
+        models_folder = Path(self._models_folder_path)
+        if not models_folder.is_dir():
+            raise argparse.ArgumentError(f'Given \'models_folder_path\' of {self._models_folder_path} is not a '
+                                         f'directory.')
```
Comment (elexisvenator) [inline modules/commands/FinishCommand.py]: is this a multiline string? what does the `f` on the second line do?

Comment (ChintanRaval) [reply modules/commands/FinishCommand.py]: indents, conforms to PEP-8

**Comment by ChintanRaval** [comment]
> > _No description provided._
> 
> ???

details already in story on JIRA

File: modules/ModelChangeDetector.py [hunk 3]
```diff
@@ -45,6 +45,10 @@ def __get_arguments(self):
         finish_command_parser.add_argument('execution_id',
                                            metavar='execution_id',
                                            help='data pipeline execution id as received using \'start\' command')
+        finish_command_parser.add_argument('model_folder_paths',
+                                           metavar='model-folder-paths',
```
Comment (waltaro) [inline modules/ModelChangeDetector.py]: Should `model-folder-paths` be snake case like `execution_id`?

Comment (ChintanRaval) [reply modules/ModelChangeDetector.py]: Thanks.. Good catch.. i mucked around with it a bit too much.. it intrinsically turns it into a variable with snake case but i was trying to be explicit with metavar and shit happened.

Comment (ChintanRaval) [reply modules/ModelChangeDetector.py]: turns out the discrepancy was actually in the `execution_id` and `model-folder-paths` was right all along.

File: modules/entities/ModelChecksumEntity.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+from modules.entities import DataPipelineExecutionEntity
+
+TABLE_NAME = 'model_checksum'
+
+
+class ModelChecksumEntity(Shared.BaseEntity):
+    __tablename__ = TABLE_NAME
+    __table_args__ = {'schema': Constants.DATA_PIPELINE_EXECUTION_SCHEMA_NAME}
+
+    id = Column('id',
```
Comment (waltaro) [inline modules/entities/ModelChecksumEntity.py]: Should we also use `PRIMARY_KEY_COL_NAME = 'id'` here as well? Similar to `DataPipelineExecutionEntity`

**Pull request merged**